swift/send-message/message-types/discord/Sources/send-discord.swift

```diff
@@ -0,0 +1,113 @@
+import AsyncHTTPClient 
+import Foundation 
+import NIO 
+import NIOFoundationCompat
+
+class DiscordMessenger: Messenger{
+    private let botToken: String
+    private let guildID: String
+    private let httpClient: HTTPClient
+
+    init(_ env_vars: [String: String]) throws {
+        guard let botToken = env_vars["BOT_TOKEN"],
+            let guildID = env_vars["GUILD_ID"] else {
+            throw MessengerError.misconfigurationError(error: "Missing environment variables.")
+        }
+        self.botToken = botToken
+        self.guildID = guildID
+        httpClient = HTTPClient(eventLoopGroupProvider: .createNew)
+    }
+
+    //Returns a recipient id given their username
+    private func getRecipientID(username: String) async throws -> String{
+        let targetURL = "https://discord.com/api/v10/guilds/\(guildID)/members/search?query=\(username)&limit=1000"
+        var request = HTTPClientRequest(url: targetURL)
+
+        request.method = .GET
+        request.headers.add(name: "Authorization", value: botToken) 
+        
+        let response:HTTPClientResponse
+        do{
+            response = try await httpClient.execute(request, timeout: .seconds(30))
+        }catch{
+            throw MessengerError.providerError(error: "Request url not responding or connection timed out")
+        }
+        if response.status != .ok {
```

Comment:
lets make a comment here that if the user doesn't exist we till get a 200, not a 404 as you might expect